api/v3/Case/Getdetails.php

```diff
@@ -155,23 +155,19 @@ function civicrm_api3_case_getdetails(array $params) {
           $categories[$grouping][] = $allTypes[] = $val['value'];
         }
       }
-      $activities = civicrm_api3('Activity', 'get', [
-        'return' => [
-          'activity_type_id', 'subject', 'activity_date_time', 'status_id',
-          'case_id', 'target_contact_name', 'assignee_contact_name',
-          'is_overdue', 'is_star', 'file_id', 'tag_id.name',
-          'tag_id.description', 'tag_id.color',
-        ],
+
+      // Get last activity.
+      $lastActivity = _civicrm_api3_case_get_activities($ids, 1, [
+        'check_permissions' => !empty($params['check_permissions']),
+        'status_id.filter' => CRM_Activity_BAO_Activity::COMPLETED,
```

Comment:
I am not sure about this logic, lets confirm the meaning of "Last Activity" requirement from the PM please.

Reply:
Filters configured properly already.